hasher_test.go

```diff
@@ -735,3 +769,73 @@ func Test_MustHashLeaf_Panic(t *testing.T) {
 		})
 	}
 }
+
+func TestMax(t *testing.T) {
```

Comment:
Added these tests for consistency with the tests developed for the `min` counterpart.